awx/main/models/__init__.py

```diff
@@ -99,6 +100,58 @@
 User.add_to_class('accessible_objects', user_accessible_objects)
 
 
+def convert_jsonfields():
+    if connection.vendor != 'postgresql':
+        return
+
+    # fmt: off
+    fields = [
+        ('main_activitystream', (
+            'deleted_actor',
+            'setting',
+        )),
+        ('main_job', (
+            'survey_passwords',
```

Comment:
I'm asking myself "will there be any bugs due to a field being _unexpectedly_ null / empty?". To some extent, yes, this may expose encrypted values (not the plain text values).

I have argued that the `survey_passwords` field is not necessary in the first place https://github.com/ansible/awx/pull/13091 and thus, we could remove it.

If someone was concerned about the security / optics of this, you could copy the `display_extra_vars` versions from my attempted removal of the field, and hopefully allow it to become completely vestigial.

Reply:
I'd argue vehemently that making any changes to these fields that do not recapitulate what has already landed in old main/0185 is out of scope.  This is surgery to make the change that is already out in the wild work more robustly.